### ISSUE TITLE
youtubeDNN target should be item id not 1 or 0

### ISSUE THREAD

**Issue by a-whitej** [shenweichen/DeepMatch#36]
**Describe the bug(问题描述)**

YoutubeDNN模型训练的时候，target应该是item的id，不是1或者0的是否点击的标识。
如下的loss调用中的label_idx应该是item的id.
        loss = tf.nn.sampled_softmax_loss(weights=embeddings,  # self.item_embedding.
                                          biases=self.zero_bias,
                                          labels=label_idx,
                                          inputs=inputs,
                                          num_sampled=self.num_sampled,
                                          num_classes=self.size,  # self.target_song_size
                                          )
在下面函数中返回的应该是train_iid ，实际返回了train_label
https://github.com/shenweichen/DeepMatch/blob/ea69d9cc9aec3742ae70f487b3ec9491fdfb14a8/examples/preprocess.py#L68

如果是1或者0，没有负样本情况下，值一直是1，一直在训练一个样本

**To Reproduce(复现步骤)**

**Operating environment(运行环境):**
 - python version [e.g. 3.4, 3.6]
 - tensorflow version [e.g. 1.4.0, 1.12.0]
 - deepmatch version [e.g. 0.1.1,]

**Additional context**

**Comment by caerusy** [comment]
同样想问在negsample为0的时候train_label不一直都为1吗，模型训练的都是正样本，这样合理吗？

**Comment by Ethan199111** [comment]
sampled_softmax_loss的确是要送具体的id,不能送one-hot或者1,0 这一点和softmax_loss是有区别的。

**Comment by ninesky110** [comment]
假如说label的类型非常大 大概10w左右，这种训练方法还适用吗

**Comment by ninesky110** [comment]
> 假如说label的类型非常大 大概10w左右，这种训练方法还适用吗

tf.nn.sampled_softmax_loss中的num_sampled来进行类别采样，控制最终的预测类别数。   应该是这样的吧？
